packages/patternfly-react/src/components/LoginPage/components/LoginCardComponents/LoginCardWithValidation.js

```diff
@@ -0,0 +1,253 @@
+    }
+  };
+
+  getModifiedProps = () => {
+    const { usernameField, passwordField } = this.props;
+    const passwordFieldWarningType = this.state.isCapsLock
+      ? 'capsLock'
+      : this.state.passwordField.warningType;
+    return {
+      usernameField: {
+        ...usernameField,
+        onChange: e => this.onInputChange(e, 'usernameField'),
+        onFocus: e => this.onInputFocus(e, 'usernameField'),
+        onBlur: e => this.onInputBlur(e, 'usernameField'),
+        onKeyPress: e => this.onKeyPress(e, 'usernameField'),
+        error: usernameField.errors[this.state.usernameField.errorType],
+        showError: this.state.usernameField.showError
+      },
+      passwordField: {
+        ...passwordField,
+        onChange: e => this.onInputChange(e, 'passwordField'),
+        onFocus: e => this.onInputFocus(e, 'passwordField'),
+        onBlur: e => this.onInputBlur(e, 'passwordField'),
+        onKeyPress: e => this.onKeyPress(e, 'passwordField'),
+        warning: passwordField.warnings[passwordFieldWarningType],
+        showWarning:
+          this.state.passwordField.isFocused && this.state.isCapsLock,
+        error: passwordField.errors[this.state.passwordField.errorType],
+        showError: this.state.passwordField.showError
+      },
+      onSubmit: e => this.onSubmit(e),
+      showError: this.state.form.showError
+    };
+  };
+
+  handleOnInputErrors = () => {
+    const { usernameField, passwordField } = this.state;
+    if (usernameField.value) {
+      !this.isUserNameValid() && this.handleOnInvalidUsername();
+    } else {
+      this.handleOnEmptyInput('usernameField');
+    }
+
+    if (passwordField.value) {
+      this.isPasswordShort() && this.handleOnPasswordTooShort();
+    } else {
+      this.handleOnEmptyInput('passwordField');
+    }
+
+    this.hideSubmitError();
+  };
+
+  isFormValid = () =>
+    !!this.state.usernameField.value &&
+    !!this.state.passwordField.value &&
+    !this.isPasswordShort() &&
+    this.isUserNameValid();
+
+  isPasswordShort = () =>
+    this.state.passwordField.value.length < this.props.passwordField.minLength;
+
+  hideSubmitError = () => {
+    this.setState({
+      form: {
+        ...this.state.form,
+        showError: false
+      }
+    });
+  };
+
+  handleOnPasswordTooShort = () => {
+    this.setState({
+      passwordField: {
+        ...this.state.passwordField,
+        errorType: 'short',
+        showError: true
+      }
+    });
+  };
+
+  handleOnInvalidUsername = error => {
+    this.setState({
+      usernameField: {
+        ...this.state.usernameField,
+        errorType: 'invalid',
+        showError: true
+      }
+    });
+  };
+
+  handleOnEmptyInput = inputType => {
+    this.setState({
+      [inputType]: {
+        ...this.state[inputType],
+        errorType: 'empty',
+        showError: true
+      }
+    });
+  };
+
+  toggleCapsLock = e => {
+    if (!this.state.passwordField.value) {
+      return;
+    }
+    e.key === 'CapsLock' &&
```

Comment:
We started tracking all keycodes for accessibility in `src/helpers.js` , `KEY_CODES` constant. I think it is best to have one constant for these.